setup-gcloud/src/setup-gcloud.ts

```diff
@@ -49,15 +49,27 @@ async function run() {
     }
 
     // write the service account key to a temporary file
-    const tmpKeyFilePath = await new Promise<string>((resolve, reject) => {
+    let tmpKeyFilePath = await new Promise<string>((resolve, reject) => {
       tmp.file((err, path, fd, cleanupCallback) => {
         if (err) {
           reject(err);
         }
         resolve(path);
       });
     });
-    await fs.writeFile(tmpKeyFilePath, Base64.decode(serviceAccountKey));
+
+    const serviceAccountFileName = core.getInput('service_account_file_name');
+
+    if (serviceAccountFileName) {
+      tmpKeyFilePath = `/tmp/${serviceAccountFileName}`;
```

Comment:
If the parameter is meant to be the full file path as per @EronWright 's suggestoin, then that needs to be reflected here.

Reply:
sure, forget this.